.circleci/config.yml

```diff
@@ -391,7 +391,44 @@ jobs:
       - run: source ${BASH_ENV} && make deps
       - run: brew install protobuf
       - run: PATH="$GOPATH/bin:${HOME}/goinstall/go/bin:$PATH" make generate-structs
-      - run: source ${BASH_ENV} && make pkg/darwin_amd64.zip
+
+      - run:
+          name: prepare ui cache
+          command: git log -n 1 --pretty=format:%H ui > /tmp/ui-sha
+
+      - restore_cache:
+          name: restore compiled ui assets
+          keys:
+          - v1-binary-ui-assets-darwin-{{ checksum "/tmp/ui-sha" }}
+      - run:
+          name: prepare ui
+          command: |
+            if [[ -f ~/caches/ui-assets/bindata_assetfs.go ]]; then
+              cp ~/caches/ui-assets/bindata_assetfs.go ./command/agent/bindata_assetfs.go
+              exit 0
+            fi
+
+            ./scripts/vagrant-linux-unpriv-ui.sh
+
+            export PATH="$GOPATH/bin:/usr/local/go/bin:$PATH"
+            source ${BASH_ENV}
+            . ~/.nvm/nvm.sh
+            cd ui && yarn install --frozen-lockfile && cd ..
+            JOBS=2 make ember-dist static-assets
+
+            mkdir -p ~/caches/ui-assets
+            cp ./command/agent/bindata_assetfs.go ~/caches/ui-assets/bindata_assetfs.go
+      - save_cache:
+          name: save compiled ui assets
+          key: v1-binary-ui-assets-darwin-{{ checksum "/tmp/ui-sha" }}
+          paths:
+          - ~/caches/ui-assets
```

Comment:
FYI - the compiled assets are the same in macos - it should be possible to reuse the linux cache for macos, assuming we can use the same cache paths.  However, CircleCI has a bug in handling symlinks, so couldn't use `/tmp` (as it's a symlink to `/private/tmp` in macOS) and couldn't really think of a nice alternative.

Reply:
It looks like we're only caching the `bindata_assetfs.go` file. Could we unpack the cache directly onto that file rather than into `/tmp` and then check for whether that's changed from what's in source to return early if it's a no-op?

Reply:
I considered it but hit some obstacles. Detecting if the compilation is a no-op isn't obvious (at least without downloading the node/yarn dependencies) and `make ember-dist` seems to do a clean compilation every time regardless.

Reply:
👍